### ISSUE TITLE
rabbitmq_reconnect_on_error parameter is useless

### ISSUE THREAD

**Issue by cwjohnston** [sensu/sensu-puppet#717]
## Description of problem

The code added in #333 places a `reconnect_on_error` parameter under the `rabbitmq` configuration scope, e.g.

```json
{
  "rabbitmq": {
    "reconnect_on_error": false,
  }
}
```

[Per the documentation](https://sensuapp.org/docs/0.29/reference/transport.html#transport-attributes), this configuration parameter actually belongs under the `transport` configuration scope, e.g.

```json
{
  "transport": {
    "name": "rabbitmq",
    "reconnect_on_error": true
  }
}
```

Further more, this parameter defaults to `false`, but Sensu defaults it to `true` as of version 0.18.

## Proposed solution

The `rabbitmq_reconnect_on_error` parameter should be renamed to `transport_reconnect_on_error` and it's default value should be changed from `false` to `true`.

For the most part, references to `rabbitmq_reconnect_on_error` parameter, e.g. in rabbitmq manifest and sensu_rabbitmq_config provider, can be removed.

Code in the transport manifest should be changed so that transport.json is written regardless of the value of the `transport_type` parameter, and the boolean value of `transport_reconnect_on_error` should be used to influence the content of transport.json

**Comment by ghoneycutt** [comment]
There is existing work for this in PR #595

**Comment by ghoneycutt** [comment]
Released in v2.19.0
